packages/react-devtools-shared/src/__tests__/ComponentSettings-test.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import ComponentsSettings from '../devtools/views/Settings/ComponentsSettings';
+
+describe('ComponentsSettings', () => {
+  let container;
+
+  beforeEach(() => {
+    // Set up a DOM element as a render target
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    // Clean up on exiting
+    unmountComponentAtNode(container);
+    container.remove();
+    container = null;
+  });
+
+  // @reactVersion >= 17
+  it('should update collapseNodesByDefault setting', () => {
+    const useContextMock = jest.fn().mockReturnValue({
+      collapseNodesByDefault: true,
+      addListener: jest.fn(),
+      removeListener: jest.fn(),
+    });
+
+    render(<ComponentsSettings />, container);
```

Comment:
Could you please try to render `ComponentsSettings` wrapper in required contexts?
Like this:

```
render(
  <StoreContext.Provider value={...}>
    <SettingsContext.Provider value={...}>
      <ComponentsSettings />
    </SettingsContext.Provider>
  </StoreContext.Provider>,
  container
);
```

I think this should resolve your problem with mocking contexts and you won't be required to mock `useContextMock`

Reply:
cc @Biki-das, have you tried this?

Reply:
@hoxyq let me try :-) , was busy with other stuff!

Reply:
the issue here is i don't have experience testing react components without rtl 😅, so i am really struggline to make this up!
Plain jest tests are hard to come up with

Reply:
There is no rush, thanks for working on this!

Reply:
@hoxyq could you try to see around the existing test file , even rendering component setting is throwing error, i feel i am somewhere not understanding to test this correctly.

Reply:
> 

I will try to take a look this week.

Reply:
@hoxyq did you looked the test files?

Reply:
Not yet, sorry, I am finishing something and then will take a look at this, it is still in my queue

Reply:
I've tried to find a workaround for these tests, but I think we will need to find a better approach on how they should be implemented.

We can remove these tests for now:
- We don't have a good example with unit tests for custom components, some tests seem outdated and cover integration with user code
- It is quite painful atm to write tests with using `dispatchEvent` because of how SyntheticEvents work

Reply:
yup that's what i felt too, when i started out, so for now we would remove the test and get this merged?

Reply:
> yup that's what i felt too, when i started out, so for now we would remove the test and get this merged?

Yes, I will review the PR after your changes and hope we can ship this soon :)